pkg/blackduck/crdinstaller.go

```diff
@@ -0,0 +1,282 @@
+	protoform.SetupHTTPServer(c.kubeClient, c.hubClient, c.config.Namespace)
+}
+
+// CreateInformer will create a informer for the CRD
+func (c *CRDInstaller) CreateInformer() {
+	c.infomer = hubinformerv2.NewBlackduckInformer(
+		c.hubClient,
+		c.config.Namespace,
+		c.resyncPeriod,
+		c.indexers,
+	)
+}
+
+// CreateQueue will create a queue to process the CRD
+func (c *CRDInstaller) CreateQueue() {
+	// create a new queue so that when the informer gets a resource that is either
+	// a result of listing or watching, we can add an idenfitying key to the queue
+	// so that it can be handled in the handler
+	c.queue = workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+}
+
+// AddInformerEventHandler will add the event handlers for the informers
+func (c *CRDInstaller) AddInformerEventHandler() {
+	c.infomer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			// convert the resource object into a key (in this case
+			// we are just doing it in the format of 'namespace/name')
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err == nil {
+				// add the key to the queue for the handler to get
+				c.queue.Add(key)
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(newObj)
+			if err == nil {
+				c.queue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// DeletionHandlingMetaNamsespaceKeyFunc is a helper function that allows
+			// us to check the DeletedFinalStateUnknown existence in the event that
+			// a resource was deleted but it is still contained in the index
+			//
+			// this then in turn calls MetaNamespaceKeyFunc
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			log.Infof("delete hub: %s: %+v", key, obj)
+
+			if err == nil {
+				c.queue.Add(key)
+			}
+		},
+	})
+}
+
+// CreateHandler will create a CRD handler
+func (c *CRDInstaller) CreateHandler() {
+
+	osClient, err := securityclient.NewForConfig(c.kubeConfig)
+	if err != nil {
+		osClient = nil
+	} else {
+		_, err := util.GetOpenShiftSecurityConstraint(osClient, "anyuid")
+		if err != nil && strings.Contains(err.Error(), "could not find the requested resource") && strings.Contains(err.Error(), "openshift.io") {
+			log.Debugf("Ignoring scc privileged for kubernetes cluster")
+			osClient = nil
+		}
+	}
+
+	routeClient, err := routeclient.NewForConfig(c.kubeConfig)
+	if err != nil {
+		routeClient = nil
+	} else {
+		_, err := util.GetOpenShiftRoutes(routeClient, "default", "docker-registry")
+		if err != nil && strings.Contains(err.Error(), "could not find the requested resource") && strings.Contains(err.Error(), "openshift.io") {
+			log.Debugf("Ignoring routes for kubernetes cluster")
+			routeClient = nil
+		}
+	}
+
+	c.handler = NewHandler(c.config, c.kubeConfig, c.kubeClient, c.hubClient, c.defaults.(*v1.BlackduckSpec), fmt.Sprint("http://federator:3016"), make(chan bool, 1), osClient, routeClient)
```

Comment:
you are right. we can cleanup the addition of federator url.. not mandatory

Reply:
The federator url is set but we do not actually do anything with it since we never call _callHubFederator_. Do you want me to remove all the federator code or just leave it the way it is?